src/Http/Http.Results/src/Results.cs

```diff
@@ -712,12 +712,33 @@ public static IResult InternalServerError<TValue>(TValue? error)
     /// <param name="extensions">The value for <see cref="ProblemDetails.Extensions" />.</param>
     /// <returns>The created <see cref="IResult"/> for the response.</returns>
     public static IResult Problem(
+        string? detail,
+        string? instance,
+        int? statusCode,
+        string? title,
+        string? type,
+        IDictionary<string, object?>? extensions)
+        => TypedResults.Problem(detail, instance, statusCode, title, type, extensions);
+
+    /// <summary>
+    /// Produces a <see cref="ProblemDetails"/> response.
+    /// </summary>
+    /// <param name="statusCode">The value for <see cref="ProblemDetails.Status" />.</param>
+    /// <param name="detail">The value for <see cref="ProblemDetails.Detail" />.</param>
+    /// <param name="instance">The value for <see cref="ProblemDetails.Instance" />.</param>
+    /// <param name="title">The value for <see cref="ProblemDetails.Title" />.</param>
+    /// <param name="type">The value for <see cref="ProblemDetails.Type" />.</param>
+    /// <param name="extensions">The value for <see cref="ProblemDetails.Extensions" />.</param>
+    /// <returns>The created <see cref="IResult"/> for the response.</returns>
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads
+    public static IResult Problem(
```

Comment:
Can you elaborate a bit on why you preferred to have two separate code paths, rather than having the new overload delegate to the old one?  I can see arguments in both directions.

Reply:
This method just delegates to `TypedResult`. In `TypedResult`, if I do `return Problem(detail, instance, statusCode, title, type, extensions);` to call the other overload, it calls the same method and creates a recursion.

Reply:
You should be able to force a particular overload with casting.

Reply:
(Not recommending you do that, just saying it doesn't seem like a compelling reason one way or the other on its own.)

Reply:
Will investigate that to avoid to duplicate the implementation.

Reply:
I avoided duplicating using a cast.